src/applications/helper/gaming-traffic-helper.cc

```diff
@@ -0,0 +1,62 @@
+/* -*- Mode:C++; c-file-style:"gnu"; indent-tabs-mode:nil; -*- */
+/*
+ * Copyright (c) 2020, University of Padova, Department of Information
+ * Engineering, SIGNET Lab.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2 as
+ * published by the Free Software Foundation;
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+ *
+ * Authors: Salman Mohebi <s.mohebi22@gmail.com>
+ *
+ */
+#include "gaming-traffic-helper.h"
+#include "ns3/udp-server.h"
+#include "ns3/udp-client.h"
+#include "ns3/udp-trace-client.h"
+#include "ns3/uinteger.h"
+#include "ns3/string.h"
+#include "ns3/gaming-streaming-server.h"
+
+namespace ns3 {
+GamingStreamingServerHelper::GamingStreamingServerHelper (std::string applicationType)
+{
+  m_factory.SetTypeId (applicationType);
+}
+
+GamingStreamingServerHelper::GamingStreamingServerHelper (std::string applicationType, Address address, uint16_t port)
+  : GamingStreamingServerHelper (applicationType)
+{
+  SetAttribute ("RemoteAddress", AddressValue (address));
+  SetAttribute ("RemotePort", UintegerValue (port));
```

Comment:
does this compile? whose attribute are you setting?

Reply:
These are attributes of `m_factory` which is an object of `ObjectFactory` class.
`ObjectFactory` later use them to create different `GamingStreamingServer` objects.

Reply:
Oh right sorry, i didn't see the `GamingStreamingServerHelper::SetAttribute` method